### DISCUSSION TITLE
how to start an app with turbo on other PORT?

### DISCUSSION THREAD

**Discussion by ranlix** [vercel/turborepo#8453] [Help] [answered]
### Summary

After upgrading to "turbo": "^2.0.3", below command doesn't work any more:
`PORT=4000 pnpm turbo run dev --filter web`

then, this nextjs app starts at PORT 3000 as default. 
<img width="335" alt="image" src="https://github.com/vercel/turbo/assets/4019372/88c068dd-8bed-481b-86ed-b4900c966950">

How to pass a custom PORT environment to certain app when starting?

**Comment by anthonyshew** [comment, marked as answer]
This is expected because of [Strict Environment Mode](https://turbo.build/repo/docs/crafting-your-repository/using-environment-variables#strict-mode). The `PORT` variable is not passed through unless you use one of `env`, `passThroughEnv`, `globalEnv`, or `globalPassThroughEnv`. I'd suggest [`globalPassThroughEnv`](https://turbo.build/repo/docs/reference/configuration#globalpassthroughenv) since you likely don't want this variable to affect caching behavior and want it to be available globally.
